fms/utils/generation.py

```diff
@@ -74,9 +180,33 @@ def generate(
     kwargs: MutableMapping[str, Any] = dict()
     kwargs["past_key_value_states"] = None
     kwargs["use_cache"] = use_cache
+    kwargs["position_ids"] = position_ids
 
-    for _ in range(max_new_tokens):
+    for i in range(max_new_tokens):
         input_ids = next_input[:, -max_seq_len:]
+
+        # prefill
+        if i == 0:
+            kwargs["mask"] = __create_prefill_mask(
+                model_input_lengths, input_ids.device
+            )
+        # decode
+        else:
+            # add 1 for generate
+            model_input_lengths = [
+                model_input_lengths + 1 for model_input_lengths in model_input_lengths
```

Comment:
I think the actual change when updating the decode mask is just to append a '1' to every row?

do we actually need to track, modify, and use this list to recreate the mask in each step .... or can we just do something like `kwargs['mask'] = torch.cat(mask, torch.ones( (mask.shape[0],) )` in the case where we have a mask?